influxdb_client/client/sql_client.py

```diff
@@ -0,0 +1,108 @@
+"""InfluxDB SQL Client."""
+from urllib.parse import urlparse
+
+from influxdb_client.client._base import _BaseSQLClient
+
+
+class SQLClient(_BaseSQLClient):
+    """
+    Implementation for gRPC+TLS client for SQL.
+
+    This class provides basic operations for interacting with InfluxDB via SQL.
+    """
+
+    def __init__(self, influxdb_client, bucket):
+        """
+        Initialize SQL client.
+
+        Unlike the previous APIs, this client is is produced for a specific
+        bucket to query against. Queries to different buckets require different
+        clients.
+
+        To complete SQL requests, a different client is used. The rest of this
+        client library utilizes REST requests against the published API.
+        However, for SQL support connections are handled over gRPC+TLS. As such
+        this client takes the host and client and creates a new client
+        connection for SQL operations.
+
+        :param influxdb_client: influxdb client
+        """
+        super().__init__(influxdb_client=influxdb_client)
+
+        from flightsql import FlightSQLClient
+        self._client = FlightSQLClient(
+            host=urlparse(self._influxdb_client.url).hostname,
+            token=self._influxdb_client.token,
+            metadata={"bucket-name": bucket},
+        )
+
+    def close(self):
+        """Close the client connection."""
+        self._client.close()
+
+    def query(self, query: str):
+        """
+        Execute synchronous SQL query and return result as an Arrow reader.
+
+        :param str, query: the SQL query to execute
+        :return: PyArrow RecordbatchReader
+
+        .. code-block:: python
+
+            with InfluxDBClient(url="http://localhost:8086", token="my-token", org="my-org") as client:
+                # Each connection is specific to a bucket
+                sql_client = client.sql_client("my-bucket")
+
+                # The returned result is a stream of data. For large result-sets users can
+                # iterate through those one-by-one to avoid using large chunks of memory.
+                with sql_client.query("select * from cpu") as result:
+                    for r in result:
+                        print(r)
+
+                # For smaller results you might want to read the results at once. You
+                # can do so by using the `read_all()` method.
+                with sql_client.query("select * from cpu limit 10") as result:
+                    data = result.read_all()
+                    print(data)
+
+                # To get you data into a Pandas DataFrame use the following helper function
+                df = data.to_pandas()
+
```

Comment:
The `close` statement is missing here. So wouldn't it be better to include the example instead of duplicating it? Not sure if it works though
```suggestion
.. literalinclude:: /examples/sql_client.py
   :language: python

```

Reply:
+1 on the close

> .. literalinclude::

I think this would work when we published the docs to read the docs at https://influxdb-client.readthedocs.io/en/stable/ however, that won't show up in the README when viewed manually or via GitHub, which is why I think they have coopy and paste examples in the past.